js/renovation/ui/check_box.tsx

```diff
@@ -0,0 +1,228 @@
+      onClick={viewModel.onWidgetClick}
+      onInactive={viewModel.onInactive}
+      onKeyDown={viewModel.onWidgetKeyDown}
+      rtlEnabled={viewModel.props.rtlEnabled}
+      tabIndex={viewModel.props.tabIndex}
+      visible={viewModel.props.visible}
+      width={viewModel.props.width}
+      {...viewModel.restAttributes} // eslint-disable-line react/jsx-props-no-spreading
+    >
+      {/* eslint-disable-next-line react/jsx-props-no-spreading */}
+      <input ref={viewModel.inputRef} type="hidden" value={`${viewModel.props.value}`} {...name && { name }} />
+      <div className="dx-checkbox-container">
+        <span className="dx-checkbox-icon" ref={viewModel.iconRef} />
+        {text && (<span className="dx-checkbox-text">{text}</span>)}
+      </div>
+      {viewModel.props.useInkRipple
+                && <InkRipple config={inkRippleConfig()} ref={viewModel.inkRippleRef} />}
+    </Widget>
+  );
+};
+
+@ComponentBindings()
+export class CheckBoxProps extends BaseWidgetProps {
+  @OneWay() activeStateEnabled?: boolean = true;
+
+  @OneWay() hoverStateEnabled?: boolean = true;
+
+  @OneWay() validationError?: object | null = null;
+
+  @OneWay() validationErrors?: object[] | null = null;
+
+  @OneWay() text?: string = '';
+
+  @OneWay() validationMessageMode?: string = 'auto';
+
+  @OneWay() validationStatus?: string = 'valid';
+
+  @OneWay() name?: string = '';
+
+  @OneWay() readOnly?: boolean = false;
+
+  @OneWay() isValid?: boolean = true;
+
+  // TODO: remove null after generator bug fix
+  @TwoWay() value?: boolean | null = false;
+
+  @OneWay() useInkRipple?: boolean = false;
+
+  @Event() onFocusIn?: (e: Event) => void;
+}
+
+export const defaultOptionRules = createDefaultOptionRules<CheckBoxProps>([{
+  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+  device: (): boolean => devices.real().deviceType === 'desktop' && !(devices as any).isSimulator(),
+  options: { focusStateEnabled: true },
+}, {
+  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+  device: (): boolean => (themes as any).isMaterial((themes as any).current()),
+  options: { useInkRipple: true },
+}]);
+
+@Component({
+  defaultOptionRules,
+  jQuery: {
+    component: BaseComponent,
+    register: true,
+  },
+  view: viewFunction,
+})
+
+export class CheckBox extends JSXComponent(CheckBoxProps) {
+  @Ref() iconRef!: HTMLDivElement;
+
+  @Ref() inkRippleRef!: InkRipple;
+
+  @Ref() inputRef!: HTMLInputElement;
+
+  @Ref() widgetRef!: Widget;
+
+  @Method()
+  focus(): void {
+    this.widgetRef.focus();
+  }
+
+  @Effect()
+  contentReadyEffect(): void {
+    const { onContentReady } = this.props;
+    // eslint-disable-next-line  @typescript-eslint/no-non-null-assertion
+    onContentReady!({ element: this.widgetRef });
+  }
+
+  onActive(event: Event): void {
+    const { useInkRipple } = this.props;
+
+    useInkRipple && this.inkRippleRef.showWave({ element: this.iconRef, event, wave: 1 });
```

Comment:
const { useInkRipple } = this.props;

    useInkRipple && this.inkRippleRef.showWave({ element: this.iconRef, event, wave: 1 });

May be extract to function wave(type: 'showWave' | 'hideWave', waveId: number)

Reply:
fixed